### PR TITLE
Fix/update cypress

### PR DESCRIPTION
**Pull request by battan40**
Type of change

    [ x] New feature
    [ x] Fix
    [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

What does this fix or Change:
* reconfigures cypress testing files to include tests by user view of app
* fleshes out existing tests to include some css testing
* adds tests for display of provider information
* adds stubbing of data
* as well, added all authors to html meta tag
* adds class names for styling as well as cypress test passing
* debugs absence of ability to return to home page by adding a home link tag to med and dr view
* generally added semi colons to test files for continuity

Neccesary checkmarks:

    [ x] All Tests are Passing
    [ x] The code will run locally

Implements/Fixes/Connected Issues:

    description closes #
    connects issue #7 #11 #12 #13

Testing Changes

    [ x] Please describe what in the world happened
* There was a need to separate the files by page viewed for scalability
* Files and tests that existed were separated 
* Additional tests were added to every single page to account for both user flow and full happy path functionality

* We still need sad path testing

Checklist:

    [x ] My code has no unused/commented out code
    [x ] My code follows the style guidelines of this project
    [x ] I have performed a self-review of my own code
    [x ] I have commented my code, particularly in hard-to-understand areas
    [ x] My changes generate no new warnings
    [ x] I have added tests that prove my fix is effective or that my feature works
    [x ] New and existing unit tests pass locally with my changes
    [x ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: cypress/integration/ContributionForm_spec.js [hunk 1]
```diff
@@ -1,31 +1,54 @@
...
-  })
+describe('ContributionForm View', () => {
+
+    beforeEach(() => {
+      cy.fixture('mockDr.json')
+        .then(mockData => {
+          cy.intercept('GET', 'https://head-to-toe-be.herokuapp.com/api/v1/medical_professionals?type=doctor&state=South Dakota', {
+            statusCode: 201,
+            delay: 100,
+            body: mockData
```
Comment (MarcelineBall) [inline cypress/integration/ContributionForm_spec.js:9]: I don't know if we need to intercept a Get request here, since we're not displaying cards on this page

File: cypress/integration/drView_spec.js [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+
+  it('Should acknowledge the presence of list of of doctors with a sub title', () => {
+    cy.get('#doctors-button').should('be.visible')
+      .get('#doctors-button').click()
+      .location('pathname').should('eq', '/doctors')
+
+      .get('.dr-sub-title').should('be.visible')
+      .get('.dr-sub-title').should('contain', 'Doctor')
+      .get('.dr-sub-title').should('have.css', 'font-family', 'opendyslexic')
+  });
```
Comment (MarcelineBall) [inline cypress/integration/drView_spec.js:27]: This tests says it 'acknowledges the presence of a list', but the test seems to only test for the subtitle. We'll probably want to change the phrasing on this test

File: cypress/integration/drView_spec.js [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+      .get('#doctors-button').click()
+      .location('pathname').should('eq', '/doctors')
+
+      .get('.provider-card').should('be.visible')
+      .get('.first-name').should('be.visible')
+      .get('.first-name').should('contain', 'sea lion')
+      .get('.last-name').should('contain', 'Drowzee')
+      .get('.address').should('contain', 'Raymon')
+    });
+
```
Comment (MarcelineBall) [inline cypress/integration/drView_spec.js:48]: This test seems to have a typo as it mentions mhp, when I think it was meant to be doctors. We also might want to add a test to this for number of children to show that multiple cards are being displayed

File: cypress/integration/drView_spec.js [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+  it('Should display rating buttons for our users to give their opinions about our listed providers', () => {
+    cy.get('#doctors-button').should('be.visible')
+      .get('#doctors-button').click()
+      .location('pathname').should('eq', '/doctors')
+
+      .get('.accept-button').should('be.visible')
+      .get('.accept-button').should('contain', 'Love')
+
+      .get('.deny-button').should('be.visible')
+      .get('.deny-button').should('contain', 'D')
```
Comment (MarcelineBall) [inline cypress/integration/drView_spec.js:59]: Contain the letter 'D' isn't a particularly specific test, we will probably need some more specificity with this one

File: cypress/integration/mainView_spec.js [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+  cy.get('.App-title').should('be.visible', 'Head To Toe')
+    .get('.App-title').should('contain', 'Head To Toe')
+    .get('.App-title').should('have.css', 'font-family', 'opendyslexic')
+  });
+
+  it('Should display the form, dr and mhp buttons for user', () => {
+    cy.get('.App-buttons').should('be.visible')
+      .get('.App-buttons').should('contain', 't')
+      .get('.App-buttons').should('have.css', 'font-family', 'opendyslexic')
+  });
```
Comment (MarcelineBall) [inline cypress/integration/mainView_spec.js:33]: Having all the buttons contain 't' doesn't really seem to confirm all of them. I think we'll probably want their own test to make sure each button exists how we want it.

File: cypress/integration/mainView_spec.js [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+  it('Should allow the user to navigate away to their chosen page upon click', () => {
+    cy.get('#doctors-button').click()
+      .location('pathname').should('eq', '/doctors')
+
+      .get('#mental-health-pros-button').click()
+        .location('pathname').should('eq', '/mental_health_professionals')
+
+      .get('#submit-button').click()
+      .location('pathname').should('eq', '/submit')
+  });
```
Comment (MarcelineBall) [inline cypress/integration/mainView_spec.js:44]: I think this is more of a router test, than a main view test

File: cypress/integration/mhpView_spec.js [hunk 7]
```diff
@@ -0,0 +1,70 @@
+describe('Mental Health Professionals Display', () => {
```
Comment (MarcelineBall) [inline cypress/integration/mhpView_spec.js:1]: Most of the comments I left for doctors are also applicable for the mhp tests

File: public/index.html [hunk 8]
```diff
@@ -7,6 +7,10 @@
     <meta name="theme-color" content="#000000" />
     <meta name="description" content="An app built with every body and heart in mind" />
     <meta name="Angela Battillo" />
+    <meta name="Marceline Ball" />
+    <meta name="Riley Martin" />
+    <meta name="Aiden McGuire" />
+    <meta name="Austin Andrade" />
```
Comment (MarcelineBall) [inline public/index.html:13]: Adding author names is great. Looking into it, I think the syntax might be more like this: <meta name="author" content="John Doe">

File: src/Containers/Doctors/Doctors.js [hunk 9]
```diff
@@ -37,8 +38,12 @@ const Doctors = () => {
...
   return(
     <>
-      <h2>Doctors here</h2>
-      <section>{allDoctors}</section>
+      <h2 className='dr-sub-title'>Doctors here</h2>
+      <section>{allDoctors}
+      <Link to='/'>
+      <button className='home-button'>Home</button>
+      </Link>
+      </section>
```
Comment (MarcelineBall) [inline src/Containers/Doctors/Doctors.js:46]: Having a home button is great! We might want to move it to the nav bar in the future

**Review by MarcelineBall — COMMENTED** [review]
Overall this looks really good. There are a lot of tests added and separated into their own testing files, great job with that. I just had a few comments in specific parts in ways we could improve this

File: cypress/integration/drView_spec.js (hunk 2 above)
Comment (battan40) [reply cypress/integration/drView_spec.js:27]: Great idea

File: cypress/integration/drView_spec.js (hunk 3 above)
Comment (battan40) [reply cypress/integration/drView_spec.js:48]: Honestly, I love the test the children idea!

File: cypress/integration/drView_spec.js (hunk 4 above)
Comment (battan40) [reply cypress/integration/drView_spec.js:59]: Its not specific.  I wanted to test the className that covers all the buttons.  Each button contains the letter d.  I could add in the already existing testing layer of testing each buttons id name to show that we know how to test each individual one.

File: cypress/integration/mainView_spec.js (hunk 5 above)
Comment (battan40) [reply cypress/integration/mainView_spec.js:33]: Same answer as above, adding the already existing testing that we have to this test that tests for css or moving its css testing to those tests are two solutions to the minimum value of testing for the population of singular letter.

File: cypress/integration/mainView_spec.js (hunk 6 above)
Comment (battan40) [reply cypress/integration/mainView_spec.js:44]: This is a router test.  

However, this is what the user should be able to see and do on this view of the page.  It establishes a user flow.

File: public/index.html (hunk 8 above)
Comment (battan40) [reply public/index.html:13]: Hey marceline, the syntax You're referencing isn't visible here.

Comment (MarcelineBall) [reply public/index.html:13]: Sorry about that! I edited the comment so it is now visible
